Teapot/MockTeapot.swift

```diff
@@ -22,10 +37,14 @@ open class MockTeapot: Teapot {
 
     override func execute(verb _: Verb, path: String, parameters _: RequestParameter? = nil, headerFields _: [String: String]? = nil, timeoutInterval _: TimeInterval = 5.0, allowsCellular _: Bool = true, completion: @escaping ((NetworkResult) -> Void)) {
         self.getMockedData(forPath: path) { json, error in
-            let response = HTTPURLResponse(url: URL(string: path)!, statusCode: 200, httpVersion: nil, headerFields: nil)
+            var mockedError = error
+            let response = HTTPURLResponse(url: URL(string: path)!, statusCode: self.statusCode.rawValue, httpVersion: nil, headerFields: nil)
             let requestParameter = json != nil ? RequestParameter(json!) : nil
 
-            let networkResult = NetworkResult(requestParameter, response!, error)
+            if self.statusCode != .ok {
+                mockedError = TeapotError.invalidResponseStatus
```

Comment:
Only invalid response status is related to error codes. We should probably remove these three and rely on the status code specific issues? `missingImage` is returned when using the image download methods, for instance, but I think getting a 404 is more than enough.